### PR TITLE
Handle timeouts possible in Khepri minority in `rabbit_db_exchange`

### PR DESCRIPTION
**Pull request by the-mikedavis**
Continuation of the changes like in https://github.com/rabbitmq/rabbitmq-server/issues/11685 and https://github.com/rabbitmq/rabbitmq-server/pull/11706 but for exchanges.

The changes for exchanges are a bit more involved than prior PRs. Here's a summary of the changes:

* `rabbit_exchange:declare/7`'s spec is updated to return `{ok, #exchange{}} | {error, timeout}` where it returned just `#exchange{}` before
    * this is safe to change since the function is only ever called through RPC in the CLI's test helpers
* Callers of `rabbit_exchange:delete/3` need some small changes to handle an `{error, timeout}` return
    * I've added a `rabbit_exchange:ensure_deleted/3` wrapper that covers the common case for callers where `ok | {error, not_found}` are handled the same way.
* Set `timeout => infinity` for calls for updating the exchange serial
    * Serial updates are always made after another database operation (for example adding or deleting an exchange)
    * Ideally we would update the serial in a transaction but this would take a larger change, so I've left it for future work.

### CODE REVIEW

**Review by michaelklishin — APPROVED** [review]

File: deps/rabbit/src/rabbit_amqp_management.erl [hunk 1]
```diff
@@ -210,9 +210,18 @@ handle_http_req(<<"PUT">>,
...
-                  Internal, XArgs, Username)
+                case rabbit_exchange:declare(
+                       XName, XTypeAtom, Durable, AutoDelete,
+                       Internal, XArgs, Username) of
+                    {ok, DeclaredX} ->
+                        DeclaredX;
+                    {error, timeout} ->
+                        throw(
+                          <<"503">>,
+                          "Could not create exchange '~ts' in vhost '~ts' "
```
Comment (ansd) [inline deps/rabbit/src/rabbit_amqp_management.erl:221]: Will this crash because `~s` expects a string an `XName`  is a tuple?

Comment (ansd) [reply deps/rabbit/src/rabbit_amqp_management.erl:221]: Should we just use `rabbit_misc:rs/1` here?

Comment (the-mikedavis) [reply deps/rabbit/src/rabbit_amqp_management.erl:221]: Ah that's a resource record. Right let's use `rabbit_misc:rs/1` for that too

File: deps/rabbit/src/rabbit_amqp_management.erl [hunk 2]
```diff
@@ -285,8 +294,15 @@ handle_http_req(<<"DELETE">>,
...
-    _ = rabbit_exchange:delete(XName, false, Username),
-    {<<"204">>, null, {PermCache, TopicPermCache}};
+    case rabbit_exchange:ensure_deleted(XName, false, Username) of
+        ok ->
+            {<<"204">>, null, {PermCache, TopicPermCache}};
+        {error, timeout} ->
+            throw(
+              <<"503">>,
+              "failed to delete exchange '~ts' due to a timeout",
+              [XNameBin])
```
Comment (ansd) [inline deps/rabbit/src/rabbit_amqp_management.erl:304]: Should we use `rabbit_misc:rs/1` here to log the vhost as well and pass `XName` as argument?

File: deps/rabbit/src/rabbit_channel.erl [hunk 3]
```diff
@@ -2566,13 +2569,22 @@ handle_method(#'exchange.declare'{exchange    = XNameBin,
...
+                                             Args,
+                                             Username) of
+                    {ok, DeclaredX} ->
+                        DeclaredX;
+                    {error, timeout} ->
+                        rabbit_misc:protocol_error(
+                          internal_error,
+                          "failed to declare exchange '~ts' in vhost '~ts' "
+                          "because the operation timed out",
+                          [XNameBinStripped, VHostPath])
```
Comment (the-mikedavis) [inline deps/rabbit/src/rabbit_channel.erl:2586]: We should be able to use `rabbit_misc:rs/1` on `ExchangeName` here too

**Pull request merged**